src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java

```diff
@@ -36,15 +38,23 @@ public void initialize(Connection connection) throws SQLException {
         }
     }
 
-    public void setTransactionIsolation(Integer transactionIsolation) {
+    public void setTransactionIsolation(
+        Supplier<Optional<Connection>> currentConnection,
+        Integer transactionIsolation
+    ) throws SQLException {
+        executeIfPresent(currentConnection, connection -> connection.setTransactionIsolation(transactionIsolation));
         this.transactionIsolation = transactionIsolation;
```

Comment:
Can transactionIsolation in connection and connection parameters can go away and differ each other?

Reply:
I don't understand the questions, but I'll describe what happens in the code here. We set the parameters. Parameters are for database connections that are not initialised yet. We need to store them for future use. Sometimes we already have a connection. Then we need to set the parameter directly. We still have to store the parameter for future use (we change connections dynamically).

Here if there's no current connection. We store a parameter.
If there's a connection and it will fail to `setTransactionIsolation`, it'll throw an exception and not store it for the future.

Does it answer the questions?

Reply:
It's a similar situation as with `StatementOperation`, which is used to defer calls to the _current_ `Statement`

Reply:
@wyrzyk thanks for explnation it covered my question.

> We set the parameters. Parameters are for database connections that are not initialised ye
Can we isolate initialized from none-initialized connections?